=== modified file 'src/gdd/gddAppTable.cc'
--- src/gdd/gddAppTable.cc	2016-05-22 03:43:09 +0000
+++ src/gdd/gddAppTable.cc	2016-09-29 16:10:59 +0000
@@ -433,8 +433,10 @@
 		}
 
 		// fprintf(stderr,"Adding DD to free_list %d\n",app);
+		attr_table[group][app].sem.lock ();
 		dd->setNext(attr_table[group][app].free_list);
 		attr_table[group][app].free_list=dd;
+		attr_table[group][app].sem.unlock ();
 	}
 	else if (attr_table[group][app].type==gddApplicationTypeNormal)
 	{

