--- a/src/cas/generic/casStrmClient.cc
+++ b/src/cas/generic/casStrmClient.cc
@@ -530,8 +530,20 @@ caStatus casStrmClient::readResponse ( epicsGuard < casClientMutex > & guard,
             pChan->getCID(), status, ECA_GETFAIL );
     }
 
+    aitUint32 elementCount = 0;
+    if (desc.isContainer()) {
+        aitUint32 index;
+        int gdds = gddApplicationTypeTable::app_table.mapAppToIndex
+            ( desc.applicationType(), gddAppType_value, index );
+        if ( gdds ) {
+            return S_cas_badType;
+        }
+        elementCount = desc.getDD(index)->getDataSizeElements();
+    } else {
+        elementCount = desc.getDataSizeElements();
+    }
     ca_uint32_t count = (msg.m_count == 0) ?
-                            (ca_uint32_t)desc.getDataSizeElements() :
+                            (ca_uint32_t)elementCount :
                              msg.m_count;
 
     void * pPayload;
@@ -659,8 +671,20 @@ caStatus casStrmClient::readNotifyResponse ( epicsGuard < casClientMutex > & gua
         return ecaStatus;
     }
 
+    aitUint32 elementCount = 0;
+    if (desc.isContainer()) {
+        aitUint32 index;
+        int gdds = gddApplicationTypeTable::app_table.mapAppToIndex
+            ( desc.applicationType(), gddAppType_value, index );
+        if ( gdds ) {
+            return S_cas_badType;
+        }
+        elementCount = desc.getDD(index)->getDataSizeElements();
+    } else {
+        elementCount = desc.getDataSizeElements();
+    }
     ca_uint32_t count = (msg.m_count == 0) ?
-                            (ca_uint32_t)desc.getDataSizeElements() :
+                            (ca_uint32_t)elementCount :
                              msg.m_count;
 
     void *pPayload;

