=== modified file 'src/libCom/osi/os/default/epicsReadline.c'
--- src/libCom/osi/os/default/epicsReadline.c	2012-08-31 21:05:14 +0000
+++ src/libCom/osi/os/default/epicsReadline.c	2015-06-30 18:48:04 +0000
@@ -14,6 +14,7 @@
 
 #define epicsExportSharedSymbols
 #include "envDefs.h"
+#include "epicsExit.h"
 #include "epicsReadline.h"
 
 #define EPICS_COMMANDLINE_LIBRARY_EPICS     0
@@ -84,6 +85,14 @@
     char    *line;
 };
 
+static enum {rlNone, rlIdle, rlBusy} rlState = rlNone;
+
+static void rlExit(void *dummy) {
+    if (rlState == rlBusy)
+        rl_cleanup_after_signal();
+}
+
+
 /*
  * Create a command-line context
  */
@@ -92,6 +101,11 @@
 {
     struct readlineContext *readlineContext;
 
+    if (rlState == rlNone) {
+        epicsAtExit(rlExit, NULL);
+        rlState = rlIdle;
+    }
+
     readlineContext = malloc(sizeof *readlineContext);
     if (readlineContext != NULL) {
         readlineContext->in = in;
@@ -124,7 +138,9 @@
     free (readlineContext->line);
     readlineContext->line = NULL;
     if (readlineContext->in == NULL) {
+        rlState = rlBusy;
         line = readline (prompt);
+        rlState = rlIdle;
     }
     else {
         line = (char *)malloc (linesize * sizeof *line);

