=== modified file 'src/libCom/test/epicsExceptionTest.cpp'
--- src/libCom/test/epicsExceptionTest.cpp	2015-02-13 18:17:24 +0000
+++ src/libCom/test/epicsExceptionTest.cpp	2015-08-27 16:34:41 +0000
@@ -35,7 +35,9 @@
 
 #if defined ( _MSC_VER )
     // some interesting bugs found in the MS implementation of new
-#   if _MSC_VER > 1310  /* this gets fixed some release after visual studio 7 we hope */
+#   if _MSC_VER >= 1900
+        static size_t unsuccessfulNewSize = numeric_limits < size_t > :: max ();
+#   elif _MSC_VER > 1310  /* this gets fixed some release after visual studio 7 we hope */
         static const size_t unsuccessfulNewSize = numeric_limits < size_t > :: max ();
 #   else
         static const size_t unsuccessfulNewSize = numeric_limits < size_t > :: max () - 100;

