--- a/setup.py
+++ b/setup.py
@@ -43,11 +43,11 @@ H5PLUGINS_DEFAULT = '/usr/local/hdf5/lib/plugin'
 if sys.platform == 'darwin':
     OMP_DEFAULT = False
 else:
-    OMP_DEFAULT = True
+    OMP_DEFAULT = False

 FALLBACK_CONFIG = {
-    'include_dirs': [],
-    'library_dirs': [],
+    'include_dirs': [os.environ['CONDA_HOME'] + '/include'],
+    'library_dirs': [os.environ['CONDA_HOME'] + '/lib'],
     'libraries': [],
     'extra_compile_args': [],
     'extra_link_args': [],
@@ -277,6 +277,7 @@ with open('README.rst') as r:
     long_description = r.read()

 # TODO hdf5 support should be an "extra". Figure out how to set this up.
+EXTENSIONS = [ext_bshuf]
 setup(
     name='bitshuffle',
     version=VERSION,
