--- a/setup.py
+++ b/setup.py
@@ -43,11 +43,11 @@ H5PLUGINS_DEFAULT = '/usr/local/hdf5/lib/plugin'
 if sys.platform == 'darwin':
     OMP_DEFAULT = False
 else:
-    OMP_DEFAULT = True
+    OMP_DEFAULT = False

 FALLBACK_CONFIG = {
-    'include_dirs': [],
-    'library_dirs': [],
+    'include_dirs': [os.environ['CONDA_HOME'] + '/include'],
+    'library_dirs': [os.environ['CONDA_HOME'] + '/lib'],
     'libraries': [],
     'extra_compile_args': [],
     'extra_link_args': [],
